framework/include/imgui_manager.hpp

```diff
@@ -42,6 +43,15 @@ namespace gvk
 		void enable_user_interaction(bool aEnableOrNot);
 		bool is_user_interaction_enabled() const { return mUserInteractionEnabled; }
 
+		/**	Get or create a ImTextureID for ImGui
+		 *	
+		 *  This function creates and caches a ImTextureID or returns an already cached ImTextureID for the provided `avk::image_sampler`
+		 *
+		 *	@param[in] aImageSampler		An image sampler, that should be rendered with ImGui
+		 *  @param[out] ImTextureID			ImGui identifier for textures
+		 */
+		ImTextureID get_or_create_texture(avk::image_sampler& aImageSampler);
```

Comment:
For passing resources, please use either `avk::resource_reference<const T>` or `avk::resource_reference<T>`  in that case (see _Auto-Vk_'s [Resource Management](https://github.com/cg-tuwien/Auto-Vk#resource-management)).

However, I think that in this case, no concrete `image_sampler` instance is even required... could that be? 
If I see it right, you could just use the following overload:
```
template <typename T>
binding_data descriptor_binding(uint32_t aSet, uint32_t aBinding, uint32_t aCount, shader_type aShaderStages = shader_type::all)
```
and call it with `T=avk::image_sampler_t`.

Reply:
Not sure I can follow: "no concrete image_sampler instance is even required". `imgui_manager::get_or_create_texture` currently creates one `DescriptorSet` per user texture (The sampler in this case). If we remove the sampler argument and simply use `avk::descriptor_binding<avk::image_sampler_t>(0u, 0u, 1u, avk::shader_type::fragment)`, the function creates a `DescriptorSet` once (without an user texture specified in the layout) and always returns that same one? 

Then how would code with multiple user textures, from i.e. ` compute_image_processing` such as 
```
ImTextureID inputTexId = imguiManager->get_or_create_texture(mInputImageAndSampler);
ImGui::Image(inputTexId, ...)
ImTextureID targetTexId = imguiManager->get_or_create_texture(mTargetImageAndSampler);
ImGui::Image(targetTexId, ...)
```
work if we don't specify to which `Sampler` we want the `DescriptorSet`? Or did I completely missunderstood your intention?

Reply:
Okay, sorry... I think, I confused myself about descriptor set layouts vs. concrete descriptor set instances - in this case, a concrete descriptor set needs to be created which, of course, need the concrete image handle! So, all good, I guess. 

But I think we should rename `get_or_create_texture` to `get_or_create_texture_id` or maybe to `get_or_create_texture_descriptor` because otherwise there could be misunderstandings in what the method does. "Creating a texture" sounds like allocating an image resource, which it does not do.  I think, I prefer a name which includes "descriptor".

Reply:
Hmm.. I agree `get_or_create_texture` seems confusing because you are not creating a texture. I think your suggestion `get_or_create_texture_descriptor` sounds good as it makes clear, what the function really does.